client/src/components/Signup/Signup.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { Footer } from '../Footer/Footer';
+import './Signup.css'
+
+export default class Signup extends Component {
+  state = {
+    category: '',
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    password_repeat: ''
+  }
+  changeHandler = (e) => {
+    const key = e.target.name;
+    const val = e.target.value;
+    this.setState({ [key]: val })
+  }
+  changeSubmit = (e) => {
+    e.preventDefault();
+    //Validation check
+    if (!/^[a-zA-Z]+$/.test(this.state.first_name)) {
+      alert("Enter a valid  name")
+      return
+    }
+    if (!/^[a-zA-Z]+$/.test(this.state.last_name)) {
+      alert("Enter a valid surname")
```

Comment:
Instead of alert we can have a separate component ErrorHandler where we can add a modal displaying the error